bodhi/models/models.py

```diff
@@ -1448,12 +1458,42 @@ def unpush(self):
         self.untag()
 
         for build in self.builds:
-            koji.tagBuild(self.candidate_tag, build.nvr, force=True)
+            koji.tagBuild(self.release.candidate_tag, build.nvr, force=True)
 
         self.pushed = False
         self.status = UpdateStatus.unpushed
         mail.send_admin('unpushed', self)
 
+    def revoke(self):
+        """ Remove pending request for this update """
+        log.debug("Revoking %s" % self.title)
+
+        if self.status is not UpdateStatus.pending:
+            raise BodhiException("Can only revoke a pending update, not "
+                                 "one that is %s" % self.status.description)
+
+        # Remove the 'pending' koji tags from this update so taskotron stops
+        # evalulating them.
+        if self.request is UpdateRequest.testing:
+            self.remove_tag(self.release.pending_testing_tag)
+        elif self.request is UpdateRequest.stable:
+            self.remove_tag(self.release.pending_stable_tag)
+
+        # If we're revoking a request we need to set the status back to what it
```

Comment:
Ah, the trick I think is that when the request got previously set to ``stable``, that code *also* sets the ``status`` to ``pending``, I think.  I guess I should double-check that.